sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResponseDiagnostics.java

```diff
@@ -0,0 +1,160 @@
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.RequestTimeline;
+import com.azure.cosmos.implementation.directconnectivity.rntbd.RntbdChannelAcquisitionTimeline;
+import com.azure.cosmos.implementation.directconnectivity.rntbd.RntbdEndpointStatistics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This represents diagnostics from store response OR from cosmos exception
+ */
+public class StoreResponseDiagnostics {
+    final static Logger logger = LoggerFactory.getLogger(StoreResponseDiagnostics.class);
+    private final String partitionKeyRangeId;
+    private final String sessionTokenAsString;
+    private final double requestCharge;
+    private final String activityId;
+    private final String correlatedActivityId;
+    private final int statusCode;
+    private final int subStatusCode;
+    private final int pendingRequestQueueSize;
+    private final int requestPayloadLength;
+    private final int responsePayloadLength;
+    private final RequestTimeline requestTimeline;
+    private final RntbdChannelAcquisitionTimeline channelAcquisitionTimeline;
+    private final int rntbdChannelTaskQueueSize;
+    private final RntbdEndpointStatistics rntbdEndpointStatistics;
+    private final int rntbdRequestLength;
+    private final int rntbdResponseLength;
+    private final String exceptionMessage;
+    private final String exceptionResponseHeaders;
+
+    public static StoreResponseDiagnostics createStoreResponseDiagnostics(StoreResponse storeResponse) {
+        return new StoreResponseDiagnostics(storeResponse);
+    }
+
+    public static StoreResponseDiagnostics createStoreResponseDiagnostics(CosmosException cosmosException) {
+        return new StoreResponseDiagnostics(cosmosException);
+    }
+
+    private StoreResponseDiagnostics(StoreResponse storeResponse) {
+        this.partitionKeyRangeId = storeResponse.getPartitionKeyRangeId();
+        this.activityId = storeResponse.getActivityId();
+        this.correlatedActivityId = storeResponse.getCorrelatedActivityId();
+        this.requestCharge = storeResponse.getRequestCharge();
+        this.sessionTokenAsString = storeResponse.getSessionTokenString();
+        this.statusCode = storeResponse.getStatus();
+        this.subStatusCode = storeResponse.getSubStatusCode();
+        this.pendingRequestQueueSize = storeResponse.getPendingRequestQueueSize();
+        this.requestPayloadLength = storeResponse.getRequestPayloadLength();
+        this.responsePayloadLength = storeResponse.getResponseBodyLength();
+        this.requestTimeline = storeResponse.getRequestTimeline();
+        this.channelAcquisitionTimeline = storeResponse.getChannelAcquisitionTimeline();
+        this.rntbdChannelTaskQueueSize = storeResponse.getRntbdChannelTaskQueueSize();
+        this.rntbdEndpointStatistics = storeResponse.getEndpointStsts();
```

Comment:
seems like a typo ```storeResponse.getEndpointStsts```

Reply:
It looks like a typo, I will fix this API name.